src/utils.rs

```diff
@@ -618,7 +702,7 @@ impl<D> StyledObject<D> {
 }
 
 macro_rules! impl_fmt {
-    ($name:ident) => {
+    ($name:ident,$format_char:expr) => {
```

Comment:
Why is this necessary? Could it be extracted into a separate commit/PR?

Reply:
Well, that's probably the part I like the least about my PR, but I didn't manage otherwise.

Basically, to apply a gradient on a given string, I need to apply a color after each character.
So I needed to get the properly formatted `StyledObject<D>.value`.

Before the PR, the object was being directly printed into the formatter buffer via `fmt::$name::fmt(&self.val, f)?;`.

Now I get back a `String` from `format!(_)` macro, which require a way to format that data as an argument to delegate printing format to the same implementation (Display, Debug...).

If there is a way to get back the formatted string without such a hack, I would prefer it.